### ISSUE TITLE
ModelMapper Suppory GraalVM Native Image

### ISSUE THREAD

**Issue by syedyusufh** [modelmapper/modelmapper#706]
Does ModelMapper support GraalVM Native Image building? If not, can you please share the plan as when it would? Thanks

Have tried to build a native image using latest Graal and Spring Boot, running into multiple issues.

cc: @chhsiao90

**Comment by nexus061** [comment]
Hi, I think we have the same problem. I found a way but it should be refined. [https://github.com/modelmapper/modelmapper/issues/701](url)

**Comment by syedyusufh** [comment]
Yes @nexus061 trying the same, as well using the GraalVM tracing agent's help as well.

**Comment by alejojarahi** [comment]
Hi, was anyone able to configure modelmapper to compile native with graalvm? @syedyusufh  @nexus061

**Comment by nexus061** [comment]
@alejojarahi, @syedyusufh I made a build, but it needs the changes to the mapper in the ProxyFactory class, to cache the proxies created at runtime and register them as reflection hints. The change is not trivial and should be refined, currently proxies do not have a predetermined name

**Comment by syedyusufh** [comment]
We shouldn't be required to add the reflection hints manually as GraalVM helps the same with tracingAgent. Have you tried generating the METADATA files for GraalVM using tracingAgent and then from the output of the tracingAgent try building the native-image?

`java -agentlib:native-image-agent=config-output-dir=./src/main/resources/META-INF/native-image -jar /path-to-your-normal-maven-built-jar/application-jar-name.jar com.xxxxxxxx.xxxxx.xxxxxxxMainClass`

**Comment by JakobStadlhuber** [comment]
Any updates how to native images (preferred Spring Native)

**Comment by syedyusufh** [comment]
@JakobStadlhuber run your application using the below metadata collector by performing all the application operations / navigations. Once the metadata is generated under src/resources/META-INF/native-image, build your native image. Now with the metadata available, modelmapper required classes will be available during the runtime

`java -agentlib:native-image-agent=config-output-dir=./src/main/resources/META-INF/native-image -jar /path-to-your-normal-maven-built-jar/application-jar-name.jar com.xxxxxxxx.xxxxx.xxxxxxxMainClass`

If ModelMapper team could release a version with native-image metadata, it would be very helpful as in reality you cannot do the above steps for 100+ applications

**Comment by abarbosa-ext** [comment]
Hi guys, I am facing the same issue, do you have any news about a release to solve it?

**Comment by WatchdataGitHub** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by syedyusufh** [comment]
@chhsiao90 , @calou, @ngocnhan-tran1996  
Could please help with modelmapper support for GraalVM native image or provide inputs as how to make it work? 
Thanks in Advance

**Comment by WatchdataGitHub** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
